### ISSUE TITLE
How to add  tick labels on custom boundary shape spines

### ISSUE THREAD

**Issue by yxy-biubiubiu** [SciTools/cartopy#1559]
### Description

As the title, how to add  tick labels on custom boundary shape spines. The command ax.gridlines(draw_labels=True) doesn't work. Is it the only way to use ax.text() to realise it ?
![image](https://user-images.githubusercontent.com/64295206/81912697-83e24800-9601-11ea-9908-295266efae2a.png)
Codes are from https://scitools.org.uk/cartopy/docs/latest/gallery/always_circular_stereo.html , The changes are as follows：

#### Code to reproduce

```
ax1.gridlines(draw_labels=True)
ax2.set_boundary(circle, transform=ax2.transAxes)
ax2.gridlines(draw_labels=True)
```

#### Traceback 

```

```

<details>
 <summary>Full environment definition</summary>

### conda list

```
```

### pip list

```
```

</details>

**Comment by stefraynaud** [comment]
A quick diagnostic shows that the boundary transform is not properly handled within the gridliner.

**Comment by stefraynaud** [comment]
I have a version that works now.
I can't commit it before PR #1530.

**Comment by stefraynaud** [comment]
Here it the result:
![bug_labels_custom_boundary_1559](https://user-images.githubusercontent.com/1941408/81939463-aae74c80-95f6-11ea-84bf-e358d122df7d.png)

**Comment by yxy-biubiubiu** [comment]
You gave me hope, but you didn't tell me how long it would be. Just like the love QAQ. I wonder how long will this feature become available.

**Comment by stefraynaud** [comment]
I will make a commit to my fork, then you can get it and install from sources.

**Comment by yxy-biubiubiu** [comment]
So kind of you. Wish you good. I'm really looking forward to it.

**Comment by stefraynaud** [comment]
My fork: https://github.com/stefraynaud/cartopy.git
The branch:  labels_on_patch_not_spines_1541

**Comment by QuLogic** [comment]
Closed by #1537
